### PR TITLE
fix: clean up disabled user collections

### LINKED ISSUE #174: fix: clean up user collections and isolation labels on disable
## Summary

Disabling a Hubarr user currently removes them from publishing scope, but it does not delete any Plex collections that Hubarr previously created for that user. The user disable path only updates local user state and, when Track All Users is off, clears that user's cached watchlist rows.

This leaves a gap where stale Plex collections can remain behind with their `hubarr:*` labels. On the next collection sync, isolation filters are rebuilt from enabled users only, so the disabled user's label may be removed from shared-user exclusions while the old collection still exists in Plex.

## Current behavior

- Single-user disable goes through `PATCH /api/users/:id` and calls `db.updateUser(...)`.
- Bulk disable goes through `POST /api/users/bulk` and calls `db.bulkUpdateUsers(...)`.
- The DB layer updates `users.enabled` and may delete `watchlist_cache` rows when Track All Users is disabled.
- No disable path calls `plex.deleteCollection(...)`.
- No disable path clears the disabled user's `plex_collections` records.
- No disable path immediately clears/reapplies isolation filter state.
- The global collection reset can remove all Hubarr-labelled collections, but there is no per-user cleanup when a user is disabled.

Relevant areas:

- `src/server/app.ts` user update and bulk update routes
- `src/server/db/users.ts` `updateUser` / `bulkUpdateUsers`
- `src/server/services.ts` `runPublishPass`, `applyIsolationFilters`, `resetCollections`
- `src/server/integrations/plex.ts` collection label, collection scan, and delete helpers

## Proposed approach

Add a service-level cleanup flow that runs only when a user transitions from enabled to disabled. The DB repository should stay focused on local state; Plex deletion belongs in `HubarrServices` because it needs Plex API calls, logging, and failure handling.

Suggested implementation shape:

1. Detect enabled-to-disabled transitions in both user routes:
   - `PATCH /api/users/:id`
   - `POST /api/users/bulk`
2. After the DB update succeeds, call a new service method such as `cleanupDisabledUserCollections(user)` or a bulk-aware `cleanupDisabledUsers(users)`.
3. In that service method:
   - Build the user's Hubarr collection label with the existing `plex.createCollectionLabel(user.username)` helper.
   - Delete DB-tracked collections for that user where `plex_collections.collection_rating_key` is present.
   - Scan all Plex libraries for collections carrying the exact user label, so orphaned or renamed collections are also caught.
   - Delete matching Plex collections with the existing `plex.deleteCollection(...)` helper.
   - Clear/delete the disabled user's local `plex_collections` rows.
   - Clear stored isolation filter state with `db.clearIsolationFilterState()`.
   - Reapply isolation filters for the currently enabled publishing users, or otherwise ensure the next publish pass cannot skip due to a stale isolation hash.

## Edge cases

- Bulk disable should attempt cleanup for every user that actually transitioned from enabled to disabled.
- One Plex cleanup failure should be logged clearly and should not prevent other selected users from being processed.
- Cleanup should match by exact Hubarr label, not by title, because collection titles can collide.
- Cleanup should scan all Plex libraries, not just the user's current configured libraries, because the user's library overrides or global defaults may have changed since the collection was created.
- Existing non-Hubarr Plex sharing filters must be preserved. The current isolation filter logic already strips/rebuilds only `hubarr:*` exclusions.
- Track All Users behavior should remain unchanged for watchlist caching: disabled users may still be tracked in the background, but disabled users should not keep published Plex collections.

## Acceptance criteria

- Disabling a user deletes that user's Hubarr-labelled movie and show collections from Plex.
- Orphaned or renamed collections with that user's exact Hubarr label are deleted too.
- Collections labelled for other Hubarr users are not deleted.
- The disabled user's `plex_collections` records are cleared or deleted locally.
- Isolation filters are recalculated so disabled-user labels are removed from Plex shared-user exclusions.
- Bulk disable performs cleanup for all selected users that transitioned from enabled to disabled.
- Cleanup failures are logged with enough context to diagnose which user/collection failed.

## Test plan

- Add a service test proving cleanup deletes DB-tracked collection rating keys for a disabled user.
- Add a service test proving cleanup scans Plex libraries and deletes exact-label orphaned collections.
- Add a service test proving cleanup does not delete collections labelled for another user.
- Add an isolation test proving cleanup clears isolation state so the next sync/reapply does not skip due to the previous hash.
- Add route or service coverage for bulk disable so cleanup runs for each transitioned user only.

🤖 Generated with Codex

### PR DESCRIPTION
**Pull request by Migz93**
## Summary

Closes #174

This PR cleans up Hubarr-owned Plex collections when a user is disabled, preventing stale user-labelled collections and stale isolation-filter state from lingering after the user leaves the publishing scope. It handles both single-user and bulk disable flows, and keeps Plex cleanup in the service layer where API calls, logging, and failure handling already live.

## Changes

- `src/server/app.ts`: detects enabled-to-disabled transitions before and after user updates. Bulk updates pre-fetch currently enabled users before `bulkUpdateUsers`, then schedule cleanup only for users that actually became disabled. Single-user updates use the existing previous-user lookup for the same transition check.
- `src/server/services.ts`: adds `cleanupDisabledUserCollections`, which deletes DB-tracked collection rating keys, scans all Plex libraries for exact matching Hubarr user labels, logs per-library/collection failures, clears local collection records for disabled users, and clears isolation filter state so later publish passes cannot skip on stale inputs.
- `src/server/db/collections.ts` and `src/server/db/index.ts`: add a focused helper to delete local Plex collection records for one user.
- `src/server/integrations/plex.ts`: treats 400/404 responses during collection deletion as already-gone success cases, while still surfacing transient/auth/server failures.
- `tests/server/disabled-user-cleanup.test.ts`: covers DB-tracked cleanup, exact-label orphan cleanup, no near-label or other-user deletion, Plex-not-configured fallback, library-list fallback, failure logging, continued processing, local row deletion, and isolation-state clearing.

## Test plan

- [x] Run `npx tsx --test tests/server/disabled-user-cleanup.test.ts` to verify disabled-user cleanup behavior and edge cases.
- [x] Run `npm run check` to verify TypeScript across client and server configs.
- [x] Run `npm run lint` to verify ESLint across the repo.
- [x] Run `npx tsx --test tests/server/*.test.ts` to verify existing server coverage still passes.

🤖 Generated with Codex

## Summary by CodeRabbit

* **New Features**
  * Collections are now cleaned up when users are disabled (single and bulk operations), with async processing and aggregated reporting.
  * Added a DB-only fallback when external integration is unavailable so local collection records are still removed.

* **Bug Fixes**
  * Treats missing/invalid remote collection items as no-op instead of failing; improves error handling and logging for deletion failures.

* **Tests**
  * Added comprehensive tests covering cleanup, fallbacks, bulk behavior, and error handling.

### CODE REVIEW

File: src/server/services.ts [hunk 1]
```diff
@@ -1925,6 +1925,155 @@ export class HubarrServices {
...
+              userId: user.id,
+              displayName: user.displayName,
+              collectionRatingKey: collection.ratingKey,
+              title: collection.title,
+              libraryId: library.key,
+              message: error instanceof Error ? error.message : String(error)
+            });
+          }
+        }
+      }
```
[the remark this answers is not in the document]
Comment (Migz93) [reply src/server/services.ts:2039]: Good catch, accepted. Updated the cleanup flow so Plex libraries and collection labels are scanned once into a label index, then each disabled user merges candidates from that index plus DB-tracked records. Added a regression test covering bulk cleanup scan counts.

Validation run:
- `npx tsx --test tests/server/disabled-user-cleanup.test.ts`
- `npm run check`
- `npm run lint`
- `npx tsx --test tests/server/*.test.ts`

🤖 Generated with Codex
